libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -198,38 +221,61 @@ AP_RCProtocol_CRSF::RFMode AP_CRSF_Telem::get_rf_mode() const
 {
     AP_RCProtocol_CRSF* crsf = AP::crsf();
     if (crsf == nullptr) {
-        return AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_UNKNOWN;
+        return AP_RCProtocol_CRSF::RFMode::RF_MODE_UNKNOWN;
     }
 
     if (!_crsf_version.pending && _crsf_version.use_rf_mode) {
-        return crsf->get_link_status().rf_mode;
+        if (_crsf_version.is_elrs) {
+            return static_cast<AP_RCProtocol_CRSF::RFMode>(uint8_t(AP_RCProtocol_CRSF::RFMode::ELRS_RF_MODE_4HZ) + crsf->get_link_status().rf_mode);
+        }
+        return static_cast<AP_RCProtocol_CRSF::RFMode>(crsf->get_link_status().rf_mode);
     } else if (_crsf_version.is_tracer) {
         return AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_250HZ;
     }
 
     /*
      Note:
-     - rf mode 2 on UARTS with DMA runs @160Hz
-     - rf mode 2 on UARTS with no DMA runs @70Hz
+     - CRSF rf mode 2 on UARTS with DMA runs @160Hz
+     - CRSF rf mode 2 on UARTS with no DMA runs @70Hz
     */
     if (get_avg_packet_rate() < 40U) {
-        // no DMA rf mode 1
+        // no DMA CRSF rf mode 1
         return AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_50HZ;
     }
     if (get_avg_packet_rate() > 120U) {
-        // DMA rf mode 2
+        // DMA CRSF rf mode 2
         return AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_150HZ;
     }
     if (get_max_packet_rate() < 120U) {
-        // no DMA rf mode 2
+        // no CRSF DMA rf mode 2
         return AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_150HZ;
     }
     return AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_50HZ;
 }
 
 bool AP_CRSF_Telem::is_high_speed_telemetry(const AP_RCProtocol_CRSF::RFMode rf_mode) const
 {
-    return rf_mode == AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_150HZ || rf_mode == AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_250HZ;
+    if (!_crsf_version.is_elrs) {
+        return rf_mode == AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_150HZ || rf_mode == AP_RCProtocol_CRSF::RFMode::CRSF_RF_MODE_250HZ;
+    }
+    return get_telemetry_rate() > 30;
+}
+
+uint16_t AP_CRSF_Telem::get_telemetry_rate() const
+{
+    if (!_crsf_version.is_elrs) {
+        return get_avg_packet_rate();
+    }
+    AP_RCProtocol_CRSF* crsf = AP::crsf();
+    if (crsf == nullptr) {
+        return get_avg_packet_rate();
+    }
+    // ELRS sends 1 telemetry frame every n RC frames
+    // the 1:n ratio is user selected
+    // RC rate is measured by get_avg_packet_rate()
+    // telemetry rate = air rate - RC rate
+    const uint16_t elrs_air_rates[8] = {0, 25, 50, 100, 150, 200, 250, 500};
```

Comment:
This needs to be where the enum is so that someone doesn't add to the enum but miss this array. Size should come from the enum

Reply:
the enum has rf modes for both CRSF and ELRS whilst this is ELRS specific, but having a single point to check is a good thing, I'll move it!